public/src/api/makeApiRequest.ts

```diff
@@ -15,10 +15,20 @@ export function makeApiRequest<T>(
       return response.data;
     })
     .catch((error) => {
+      if (error.response.status === 500) {
+        return {
+          ...error,
+          errorText: 'Authentication failed, please try again later'
+        };
+      }
```

Comment:
How does 500 HTTP status connect to auth failures? Please adjust the error message and put it as a default response in all other cases